elastic/kibana
# POC: Kibana React Error Boundary

**Closed: replaced with https://github.com/elastic/kibana/pull/169010**

## Summary

Address https://github.com/elastic/kibana/issues/167159

Technical doc [Elastic internal] https://docs.google.com/document/d/1kVD3T08AzLuvRMnFrXzWd6rTQWZDFfjqmOMCoXRI-14/edit

This PR creates the `KibanaErrorBoundary` component and `KibanaErrorService` utility, and shows it in use in two main areas that control lazy-loaded UI elements:

1. Stack Management. This PR handles the error that could occur when navigating across management apps, and async code needed to render the app failed to load.
    <img width="1835" alt="rollup jobs error" src="https://github.com/elastic/kibana/assets/908371/021ea4df-e677-4499-8b1b-41120bda9e3d">

2. Share Menu Manager. This PR handles the error that could occur when the Share Menu Manager calls on a ShareMenuItem to render itself, and async code that is needed fails to load.

### Testing
1. Use a script proxy in between the browser and the Kibana server. Try https://github.com/tsullivan/simple-node-proxy
2. Script the proxy to send 404 responses for the Reporting plugin bundle, and for a bundle of some Management app.
3. Try the Share > CSV menu in Discover. It should be blocked, and handled with a toast message. Buttons in the toast should work.
4. Try the management app. It should be blocked, and handled with an EuiCallout. Refresh button and EuiAccordion should work.

### Checklist

Delete any items that are not applicable to this PR.
